source/Calamari/Kubernetes/ResourceStatus/ResourceStatusReportExecutor.cs

```diff
@@ -78,12 +78,7 @@ public void ReportStatus(string workingDirectory, ICommandLineRunner commandLine
                 ? new InfiniteTimer(TimeSpan.FromSeconds(PollingIntervalSeconds)) as ITimer
                 : new Timer(TimeSpan.FromSeconds(timeoutSeconds), TimeSpan.FromSeconds(PollingIntervalSeconds));
 
-            var completedSuccessfully = statusChecker.CheckStatusUntilCompletionOrTimeout(definedResources, timer, kubectl, new Options() {  WaitForJobs = waitForJobs});
-
-            if (!completedSuccessfully)
-            {
-                throw new TimeoutException("Not all resources have deployed successfully within timeout");
```

Comment:
We still need to throw the exception, otherwise the status code of Calamari wqill be zero right?

Reply:
Fixed this. But I think it is better to communicate whether it has been successful using return value rather than exceptions. So I made this method to return a `bool`.